hashes/src/sha256.rs

```diff
@@ -71,31 +71,33 @@ impl HashEngine {
             buffer: [0; BLOCK_SIZE],
         }
     }
-}
-
-impl Default for HashEngine {
-    fn default() -> Self { Self::new() }
-}
-
-impl crate::HashEngine for HashEngine {
-    type MidState = Midstate;
 
+    /// Outputs the midstate of the hash engine. This function should not be
+    /// used directly unless you really know what you're doing.
```

Comment:
Expand and format the doc, please?

Reply:
This PR is a move-and-remove-only change. Would it be ok to do this in #3010 instead where we've had more discussion on this point?

Reply:
Oh, I've confused myself, yeah.